src/ILCompiler.RyuJit/src/JitInterface/CorInfoImpl.RyuJit.cs

```diff
@@ -484,5 +485,109 @@ private InfoAccessType constructStringLiteral(CORINFO_MODULE_STRUCT_* module, md
             ppValue = (void*)ObjectToHandle(stringObject);
             return stringObject.RepresentsIndirectionCell ? InfoAccessType.IAT_PVALUE : InfoAccessType.IAT_VALUE;
         }
+
+        private void setVars(CORINFO_METHOD_STRUCT_* ftn, uint cVars, NativeVarInfo* vars)
+        {
+            if (_localSlotToInfoMap == null && _parameterIndexToNameMap == null)
+            {
+                return;
+            }
+
+            uint paramCount = (_parameterIndexToNameMap == null) ? 0 : (uint)_parameterIndexToNameMap.Count;
+            Dictionary<uint, DebugVarInfo> debugVars = new Dictionary<uint, DebugVarInfo>();
+            int i;
+
+            for (i = 0; i < cVars; i++)
```

Comment:
I don't understand why "i" cannot be defined within the for loop using the "for (int i = 0...)" syntax - I see no break statements and the variable gets reassigned after the loop finishes - what am I missing?

Reply:
This was copy / pasted from corinfoimpl.cs. I can clean it up a bit, sure.

Reply:
It looks like 'i' is reused in the 'foreach' statement below.